jades/jades_validator.go

```diff
@@ -0,0 +1,125 @@
+package jades
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"io"
+	"net/http"
+
+	"github.com/fiware/VCVerifier/common"
+	"github.com/fiware/VCVerifier/logging"
+)
+
+const TOKEN_EXTRACTION_STRATEGY = "NONE"
+const DOCUMENT_NAME = "RemoteDocument"
+
+var ErrorBadResponse = errors.New("bad_response_from_validation_endpoint")
+var ErrorEmptyBodyResponse = errors.New("empty_body_response_from_validation_endpoint")
+var ErrorValidationServiceNotReady = errors.New("validation_service_not_ready")
+
+// Validator for JAdES(https://www.etsi.org/deliver/etsi_ts/119100_119199/11918201/01.01.01_60/ts_11918201v010101p.pdf) signatures
+type JAdESValidator interface {
+	ValidateSignature(signature string) (bool, error)
```

Comment:
Is that only the signature or the whole signed document?

Reply:
The whole singed document. Its called signature, since from a spec point of view its a "jws"